connectors/huggingface_connecter.py

```diff
@@ -0,0 +1,77 @@
+from datasets import load_dataset
+from nomic import AtlasDataset
+from ulid import ULID
+import numpy as np
+
+# Gets data from HF dataset
+def get_hfdata(dataset_identifier):
+    try:
+        # Loads dataset without specifying config
+        dataset = load_dataset(dataset_identifier)
+    except ValueError as e:
+        # Handles error messages
+        error_message = str(e)
+        if "Please pick one among the available configs" in error_message:
+            try:
+                available_configs_start = error_message.index("['") + 2
+                available_configs_end = error_message.index("']")
+                available_configs = error_message[available_configs_start:available_configs_end].split("', '")
+                dataset = load_dataset(dataset_identifier, available_configs[0], trust_remote_code=True)
+            except ValueError:
+                raise ValueError("Failed to get available configurations")
+        else:
+            raise e
+
+
+    # Processes dataset entries
+    data = []
+    for split in dataset.keys():
```

Comment:
same question here about large datasets - are we sure this will not break?

Reply:
Could I solve this with batch processing and by passing `streaming=True` through load dataset? Aaron mentioned that so it seems like that could prevent that issue